### PR TITLE
Small optimisations

### PR DESCRIPTION
**Pull request by etiennesegonzac**

### CODE REVIEW

**Comment by etiennesegonzac** [comment]
Will rebase on top of #2

File: poplar.js [hunk 1]
```diff
@@ -115,8 +118,22 @@ function elementify(html) {
...
+
+  var parts = path.split('.');
+
+  // Fast paths
+  if (parts.length == 1) {
+    return item[parts[0]];
+  }
+  if (parts.length == 2) {
+    return item[parts[0]][parts[1]];
+  }
```
Comment (wilsonpage) [inline poplar.js]: NIT: Single line `if` look cleaner to me. Might just be personal preference though :)

``` js
if (!path) return;
var parts = path.split('.');

// Fast paths
if (parts.length == 1) return item[parts[0]];
if (parts.length == 2) return item[parts[0]][parts[1]];

return getDeep(item, parts);
```

Comment (wilsonpage) [inline poplar.js]: We're not in Gaia land so we can break all the rules ;)

Comment (wilsonpage) [inline poplar.js]: BTW nice optimizations :)

File: poplar.js [hunk 2]
```diff
@@ -63,18 +63,21 @@ function poplar(html) {
 poplar.populate = function(el, data) {
   var bindings = elements.get(el);
 
-  bindings.textNodes.forEach(function(binding) {
+  for (var i = 0; i < bindings.textNodes.length; i++) {
+    var binding = bindings.textNodes[i];
     var value = getProp(data, binding.key);
     binding.textNode.data = value;
-  });
+  };
```
Comment (wilsonpage) [inline poplar.js]: NIT no `;` required, would fail linting.

File: poplar.js [hunk 3]
```diff
@@ -63,18 +63,21 @@ function poplar(html) {
...
+  // TODO: might not be roc-y enough ;)
+  for (var i = 0; i < bindings.attributes.length; i++) {
+    var binding = bindings.attributes[i];
     var value = binding.template.replace(regex.variable, function(match, group) {
       return getProp(data, group);
     });
 
     binding.el.setAttribute(binding.attribute, value);
-  });
+  };
```
Comment (wilsonpage) [inline poplar.js]: NIT no `;` required, would fail linting.

**Pull request merged**